### ISSUE TITLE
Books module

### ISSUE THREAD

**Issue by manparvesh** [swapagarwal/JARVIS-on-Messenger#160]
The books module does not seem to be working properly. Some of the queries didn't return any response at all. Please take a look.
Here are some of the queries I used:
![screenshot_20170124-130716](https://cloud.githubusercontent.com/assets/6629115/22234855/56fc1dac-e236-11e6-911f-9b95bc2f350b.png)

**Comment by abhisheksp** [comment]
There seems to be no response or delayed response from the goodreads API.

Some of the things I observed in the codebase:
1. The units tests are only testing only a minute part of the actual functionality. Issues like these can be caught way in advance with mock API servers with delayed response or no response.
2. a lot of refactor opportunities. 

I would love to do some refactoring and improve the unit tests :)

**Comment by swapagarwal** [comment]
@abhisheksp I would love to incorporate those! 😄  Would you still be willing to work on this?

**Comment by abhisheksp** [comment]
@swapagarwal Yes i'll handle it. Since I am busy with some personal stuff, it'll be a bit delayed.

**Comment by swapagarwal** [comment]
@abhisheksp Sure, np!
